input/postgres/statements.go

```diff
@@ -117,10 +101,59 @@ func GetStatements(server *state.Server, logger *util.Logger, db *sql.DB, global
 				" the monitoring helper functions (https://github.com/pganalyze/collector#setting-up-a-restricted-monitoring-user)" +
 				" or connect as superuser, to get query statistics for all roles.")
 		}
+
+		var extSchema string
+		var foundExtVersion float32
+		err = db.QueryRow(QueryMarkerSQL+"SELECT nspname, extversion FROM pg_extension pge INNER JOIN pg_namespace pgn ON pge.extnamespace = pgn.oid WHERE pge.extname = 'pg_stat_statements'").Scan(&extSchema, &foundExtVersion)
+		if err != nil && err != sql.ErrNoRows {
+			return nil, nil, nil, err
+		}
+
+		var extVersion float32 // defined in pg_stat_statements.c
```

Comment:
Hmm. I wonder a little bit if we can avoid treating the version as a float - and rather just use it as a text column. That appears how Postgres is treating it - i.e. versions are not meant to be compared based on their numeric value.

The branch below could then work like this, with explicitly listed version strings:

```golang
switch foundExtVersion {
case "1.0", "1.1":
  optionalFields = statementSQLDefaultOptionalFields
case "1.2":
  optionalFields = statementSQLpg94OptionalFields
case "1.3", "1.4", "1.5", "1.6", "1.7":
  optionalFields = statementSQLpg95OptionalFields
default: // 1.8 and newer
  optionalFields = statementSQLpg13OptionalFields
}
```

Reply:
It's true that `pg_extension.extversion` is a string, but that's likely just to allow third party developers to use whatever version format they want. This extension has always used a float for its version number AFAICT.

Despite that, I'm not opposed to treating them as strings -- if Postgres changes versioning scheme at some point in the future for this extension that would prevent a runtime error here. But then how would we implement the `pg_stat_statements extension outdated` log message if we can't compare them as if they were floats?

Reply:
For the outdated message, I think we could just fill in the missing versions, like this:

```golang
if postgresVersion.Numeric >= state.PostgresVersion14 {
  extVersion = "1.9"
} else if postgresVersion.Numeric >= state.PostgresVersion13 {
  extVersion = "1.8"
} else if postgresVersion.Numeric >= state.PostgresVersion12 {
  extVersion = "1.7"
} else if postgresVersion.Numeric >= state.PostgresVersion11 {
  extVersion = "1.6"
} else if postgresVersion.Numeric >= state.PostgresVersion10 {
  extVersion = "1.6"
} else if postgresVersion.Numeric >= state.PostgresVersion96 {
  extVersion = "1.4"
} else if postgresVersion.Numeric >= state.PostgresVersion95 {
  extVersion = "1.3"
} else if postgresVersion.Numeric >= state.PostgresVersion94 {
  extVersion = "1.2"
} else {
  extVersion = "1.1"
}
```

My caution here btw comes from Postgres itself not making any guarantees about this, despite the versioning scheme clearly having a pattern. It seems that when Postgres does version comparisons, it looks at the version upgrade scripts only (e.g. "pg_stat_statements--1.5--1.6.sql" indicates you can upgrade from 1.5 to 1.6) and doesn't consider the numeric sorting to be relevant.

Reply:
When I brought up the outdated message, I meant the if condition that decides whether to print it:
```go
if foundExtVersion < extVersion {

Reply:
Ah - but for that I think we can just do an equality comparison, right?

Reply:
That would cause us to incorrectly print the message if the `foundExtVersion` is higher than what their version of the collector knows to expect from Postgres. That is: they upgraded Postgres and the extension already, but not the collector.

IMO this isn't worth the effort; it's unlikely Postgres is going to change the version format out of the blue and even if they did we'd likely be releasing a new version of the collector to support new features regardless.

Reply:
Fair - we could limit the check to known Postgres versions, but agreed that this ends up being a whole lot more complicated.

I'm okay with keeping the float check for now, and adjusting if needed.